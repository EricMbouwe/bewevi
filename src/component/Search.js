import PropTypes from 'prop-types';
import searchIcon from '../assets/search.svg';

const Search = ({
  handleChangeInput, handleClick, searchInput,
}) => (
  <div className="search main">
    <input
      className="search search-input"
      type="text"
      onChange={handleChangeInput}
      placeholder="Search"
      value={searchInput}
    />
    <button
      className="search search-button"
      type="button"
      onClick={handleClick}
    >
      <img className="search search-icon" src={searchIcon} alt="Search icon to search" />
    </button>
  </div>

);

Search.propTypes = {
  handleChangeInput: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired,
  searchInput: PropTypes.string.isRequired,

};

export default Search;
