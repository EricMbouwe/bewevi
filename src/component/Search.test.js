import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

import userEvent from '@testing-library/user-event';
import Search from './Search';

let handleChangeInput;
let handleClick;
let searchInput;
let renderReadyComponent;

beforeEach(() => {
  handleChangeInput = jest.fn();
  handleClick = jest.fn();
  searchInput = 'foobar';
  renderReadyComponent = (
    <Search
      handleChangeInput={handleChangeInput}
      handleClick={handleClick}
      searchInput={searchInput}
    />
  );
});

afterEach(() => {
  handleChangeInput.mockClear();
  handleClick.mockClear();
});

describe('<Search />', () => {
  it('renders an input with given searchInput value', () => {
    render(
      renderReadyComponent,
    );
    const searchBox = screen.getByPlaceholderText('Search');
    expect(searchBox.value).toBe(searchInput);
  });

  it('is triggering when search button is clicked', () => {
    render(
      renderReadyComponent,
    );
    expect(screen.getByText(/Search/i)).toBeInTheDocument();
    userEvent.click(screen.getByText(/Search/i));
    expect(handleClick.mock.calls.length).toBe(1);
  });

  it('is triggering handleChangeInput when there is a change of vale of input', () => {
    render(
      renderReadyComponent,
    );
    ReactTestUtils.Simulate.change(screen.getByPlaceholderText('Search'));
    expect(handleChangeInput.mock.calls.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      renderReadyComponent,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
