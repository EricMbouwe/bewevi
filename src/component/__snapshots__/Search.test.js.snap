// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<div
  className="search main"
>
  <input
    className="search search-input"
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value="foobar"
  />
  <button
    className="search search-button"
    onClick={[MockFunction]}
    type="button"
  >
    <img
      alt="Search icon to search"
      className="search search-icon"
      src="search.svg"
    />
  </button>
</div>
`;
